File: src/main/kotlin/top/xiamuyao/fastspringdevelop/util/RetResult.java
package top.xiamuyao.fastspringdevelop.util;

import top.xiamuyao.fastspringdevelop.enums.ResultCode;

/**
 * ================================================
 * 作    者：夏沐尧  Github地址：https://github.com/XiaMuYaoDQX
 * 版    本：1.0
 * 创建日期： 2018/9/26
 * 描    述：
 * 修订历史：
 * ================================================
 */
public class RetResult<T> {
    /**
     * 错误码
     */
    public int code;

    /**
     * 提示信息
     */
    private String msg;

    /**
     * 具体内容
     */
    private T data;

    public RetResult<T> setCode(ResultCode retCode) {
        this.code = retCode.code;
        return this;
    }

    public int getCode() {
        return code;
    }

    public RetResult<T> setCode(int code) {
        this.code = code;
        return this;
    }

    public String getMsg() {
        return msg;
    }

    public RetResult<T> setMsg(String msg) {
        this.msg = msg;
        return this;
    }

    public T getData() {
        return data;
    }

    public RetResult<T> setData(T data) {
        this.data = data;
        return this;
    }

}

File: src/main/kotlin/com/confluence/fast/util/RetResult.kt
package com.confluence.fast.util

import com.confluence.fast.enums.ResultCode

/**
 * ================================================
 * 作    者：夏沐尧  Github地址：https://github.com/XiaMuYaoDQX
 * 版    本：1.0
 * 创建日期： 2018/9/26
 * 描    述：
 * 修订历史：
 * ================================================
 */
class RetResult<T> {
    /**
     * 错误码
     */
    var errorCode = 0
    /**
     * 提示信息
     */
    var errorMessage: String? = null
        private set
    /**
     * 具体内容
     */
    var data: T? = null
        private set
    /**
     * 总数
     */
    var total =0L
    /**
     * 当前页数
     */
    var current = 0L

    fun setTotal(total: Long): RetResult<T> {
        this.total = total
        return this
    }

    fun setCurrent(current: Long): RetResult<T> {
        this.current =current
        return this
    }

    fun setCode(retCode: ResultCode): RetResult<T> {
        errorCode = retCode.code
        return this
    }

    fun setCode(code: Int): RetResult<T> {
        this.errorCode = code
        return this
    }

    fun setMsg(msg: String?): RetResult<T> {
        this.errorMessage = msg
        return this
    }

    fun setData(data: T): RetResult<T> {
        this.data = data
        return this
    }

}